GoogleUtilities/ISASwizzler/GULObjectSwizzler.m

```diff
@@ -123,11 +128,20 @@ - (nullable id)getAssociatedObjectForKey:(NSString *)key {
 
 - (void)swizzle {
   __strong id swizzledObject = _swizzledObject;
+
+  GULObjectSwizzler *existingSwizzler =
+      [[self class] getAssociatedObject:swizzledObject key:kSwizzlerAssociatedObjectKey];
+  if (existingSwizzler != nil) {
+    NSAssert(existingSwizzler == self, @"The swizzled object has a different swizzler.");
+    // The object has been swizzled already.
+    return;
+  }
+
   if (swizzledObject) {
     [GULObjectSwizzler setAssociatedObject:swizzledObject
                                        key:kSwizzlerAssociatedObjectKey
                                      value:self
-                               association:GUL_ASSOCIATION_RETAIN_NONATOMIC];
+                               association:GUL_ASSOCIATION_RETAIN];
```

Comment:
Why is this atomic? Were there any thread safety issues we had on this earlier?

Reply:
No, there were no thread safety issues observed by me. I added it because technically the  associated object may be accessed from different threads. E.g. invoking a method on some swizzled objects implies access to the associated object, see [here](https://github.com/firebase/firebase-ios-sdk/blob/296e5504ac6dd27baf7d04a4f7093fb53092e4b0/GoogleUtilities/ISASwizzler/GULSwizzledObject.m#L60). Though I recognize that atomic may introduce additional performance cost,  I would prefer to keep it atomic since we don't have control over the swizzled object life cycle and it's better to be slower than e.g. crashing. WDYT?